### DISCUSSION TITLE
Switch off automatic workflow runs on GitHub

### DISCUSSION THREAD

**Discussion by elsie-h** [opensafely/documentation#733] [Q&A] [answered]
Hey,

The workflow runs for my project take >6 hours to run, and as the repo is private (the work is not yet published), the workflow runs cost money. I manually cancel the runs if I don't need them, but @bloodearnest and @iaindillingham mentioned that it is possible to switch off automatic workflow runs on GitHub. Could you please let me know how this is done, and if it is recommended?

Thanks!

**Comment by iaindillingham** [comment, marked as answer]
Hello @elsie-h, thanks for your question.

Workflows are specified in YAML files in your study's `.github/workflows` directory (the `.` isn't a typo; "dot" directories are hidden on macOS and Linux systems). If you created your study from the [research-template][], then the workflow is `.github/workflows/test_runner.yaml`.

I'd suggest changing

```yaml
on: [push, workflow_dispatch]
```

to

```yaml
on: [workflow_dispatch]
```

Doing so will mean the workflow is no longer run when you push local changes to GitHub; it is only run when you click _Run workflow_, in the UI.

![Screenshot 2022-05-19 at 15-15-49 Actions · opensafely_research-template](https://user-images.githubusercontent.com/477263/169315431-f1509945-7331-4833-bbbe-68120f582e43.png)

The [GitHub Actions Documentation][] has more information about the YAML files.

[research-template]: https://github.com/opensafely/research-template
[GitHub Actions Documentation]: https://docs.github.com/en/actions

**Reply by elsie-h** [reply]
Thanks @iaindillingham!
